azure/firewall/variables.tf

```diff
@@ -94,5 +94,5 @@ variable "firewall_policy_id" {
 variable "firewall_zones" {
   description = "Specifies a list of Availability Zones in which this Azure Firewall should be located."
   type        = list(string)
-  default     = null
+  default     = [1, 2, 3]
```

Comment:
We can derive this variable's value based on location:

```
data "azurerm_location" "main" {
  location = "uksouth"
}
locals {
  zones = [ for zone in data.azurerm_location.main.zone_mappings : zone.logical_zone ]
}
```

Reply:
Unless you can think of a reason why we wouldn't want to use the maximum number of zones available to us?

Reply:
Added zone_redundant bool and set default to true
Used the described code and set zones to var.zone_redundant ? local.zones : []